src/scipp/core/comparison.py

```diff
@@ -165,6 +166,12 @@ def identical(x: VariableLike,
         True if x and y have identical values, variances, dtypes, units,
         dims, shapes, coords, and masks. Else False.
     """
+    if isinstance(x, data_group.DataGroup):
```

Comment:
So this function doesn't work with `data_group_nary` because it has the special `if x.keys() != y.keys():` branch?

Reply:
`data_group_nary` always returns a `DataGroup`. Here we want to apply `all` over items and return `bool`.